File: baekjoon/8주차/2293-동전-1/우섭/2293.py
from sys import stdin
n, k = map(int, stdin.readline().split())
coin = []
for _ in range(n):
    coin.append(int(stdin.readline()))
dp = [0 for _ in range(k+1)]
dp[0] = 1
for i in coin:
    for j in range(i, k+1):
        dp[j] += dp[j - i]
print(f"{dp[k]}")

File: baekjoon/8주차/2293-동전-1/창섭/2293.kt
import java.util.*

fun main() = with(System.`in`.bufferedReader()) {
    val t = readLine()
    val n = Integer.parseInt(t.split(" ")[0])
    val k = Integer.parseInt(t.split(" ")[1])
    var coin = IntArray(n)
    var dp = IntArray(k+1)
    for (i in 0 until n){
        coin[i] = readLine().toInt()
        dp[coin[i]] = 1
    }
    for(i in 2..k){
        for (j in 0 until n){
            if(i-coin[j]>0)
                dp[i] += dp[i-coin[j]]
        }
    }
    println(dp[k])


}